Sources/NIOSSL/SSLCertificate.swift

```diff
@@ -110,6 +110,35 @@ public class NIOSSLCertificate {
         self.init(withReference: ref!)
     }
 
+    /// Create a NIOSSLCertificate from a buffer of bytes in either PEM or DER format.
+    internal convenience init(bytes ptr: UnsafeRawBufferPointer, format: NIOSSLSerializationFormats) throws {
+        // TODO(cory):
+        // The body of this method is exactly identical to the initializer above, except for the "withUnsafeBytes" call.
+        // ContiguousBytes would have been the lowest effort way to reduce this duplication, but we can't use it without
+        // bringing Foundation in. Probably we should use Sequence where Element == UInt8 and the withUnsafeContiguousBytesIfAvailable
+        // method, but that's a much more substantial refactor. Let's do it later.
+        let bio = CNIOBoringSSL_BIO_new_mem_buf(ptr.baseAddress, CInt(ptr.count))!
+
+        defer {
+            CNIOBoringSSL_BIO_free(bio)
+        }
+
+        let ref: UnsafeMutablePointer<X509>?
+
+        switch format {
+        case .pem:
+            ref = CNIOBoringSSL_PEM_read_bio_X509(bio, nil, nil, nil)
+        case .der:
+            ref = CNIOBoringSSL_d2i_X509_bio(bio, nil)
+        }
+
+        if ref == nil {
+            throw NIOSSLError.failedToLoadCertificate
+        }
+
+        self.init(withReference: ref!)
```

Comment:
the memory ownership here looks interesting. This can only be correct if `CNIOBoringSSL_PEM_read_bio_X509` and `CNIOBoringSSL_d2i_X509_bio` return a +1'd `ref` that is valid without the `bio`. If that's the case, I'd like to rename the `withReference` argument to `takingOwnership` to make this 100% clear.